sdk/src/signature.rs

```diff
@@ -30,6 +30,22 @@ impl Signature {
     }
 }
 
+pub trait Signable {
+    fn sign(&mut self, keypair: &Keypair) {
+        let data = self.get_sign_data();
+        self.set_signature(Signature::new(&keypair.sign(&data).as_ref()));
+    }
+    fn verify(&self) -> bool {
+        self.get_signature()
+            .verify(&self.pubkey().as_ref(), &self.get_sign_data())
+    }
+
+    fn pubkey(&self) -> Pubkey;
+    fn get_sign_data(&self) -> Vec<u8>;
```

Comment:
naming nit... getters don't have get_ on them in our conventions

Reply:
I didn't come up with that name. Txs already use it :P Suggestions? Just sign_data (could be confused with "please sign this data")?

Reply:
could be "signed_data()" ?

Reply:
Agreed. I'll update. Thanks!

Reply:
signable_data() ?

Reply:
yup, awesome. Thanks!